nuxeo-features/rest-api/nuxeo-rest-api-server/src/main/java/org/nuxeo/ecm/restapi/server/jaxrs/PaginableObject.java

```diff
@@ -57,6 +59,8 @@ protected void initialize(Object... args) {
 
         final HttpServletRequest request = ctx.getRequest();
         currentPageIndex = extractLongParam(request, "currentPageIndex", 0L);
+        String offsetParam = request.getParameter("offset");
+        offset = offsetParam == null ? null : Long.parseLong(offsetParam);
```

Comment:
i am wondering if we can have an empty offset, because it will not be null and it will try to parse it and it will fail `NumberFormatException`

Reply:
Looking at this https://github.com/nuxeo/nuxeo/blob/master/nuxeo-services/nuxeo-platform-query-api/src/main/java/org/nuxeo/ecm/platform/query/core/PageProviderServiceImpl.java#L137-L139 it seems that null is handled 🙂

Reply:
yep, perhaps i missed something, but my question is:

As request is a httpRequest and offsetParam a query parameter i asked if the client can send a thing like `......&offsetParam=""` and the `request.getParameter("offset");` will give `""` and it will fail when trying to parse ... or perhaps we don't care about this case because the client send something wrong :)

Reply:
Sorry, I misunderstood your question!
Not sure what is the approach for that but judging by the other parameters we could use the `extractLongParam` and default it to zero.
WDYT?
@guirenard

Reply:
Sending an empty string for the `offset` param client-side end up with `request.getParameter("offset");` returning null so it is OK.

You can quickly figure it up by tweaking the unit test https://github.com/nuxeo/nuxeo/pull/3613/files#diff-f04a983123801367a7bc6f429e88611dR436